lib/custom_wizard/extensions/topic_extension.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module CustomWizardTopicExtension
+  extend ActiveSupport::Concern
+
+  included { before_validation :check_wizard_replacement, on: :create }
+
+  def check_wizard_replacement
+    if wizard_replacing_composer?(self.category_id)
+      self.errors.add(
+        :base,
+        I18n.t('wizard.error_messages.wizard_replacing_composer')
+      )
+    end
+  end
+
+  def wizard_replacing_composer?(category_id)
+    return false unless category_id
+
+    category = Category.find(category_id)
```

Comment:
The category model instance is already available on both the post and topic models so we don't need to look it up again here.

Reply:
done